core/modules/server/authenticators/basic.js

```diff
@@ -60,35 +61,290 @@ Returns true if the request is authenticated and assigns the "authenticatedUsern
 Returns false if the request couldn't be authenticated having sent an appropriate response to the browser
 */
 BasicAuthenticator.prototype.authenticateRequest = function(request,response,state) {
-	// Extract the incoming username and password from the request
+
+    cleanAuthfile();
+	const uuid  = require('uuid');
+
+	var date = new Date();
+	//Token is invalid after 24h
+	date.setDate(date.getDate() + 1);
+	var timestamp = date.getTime();
+
+	var id=uuid.v4();
+
+	var cookies = parseCookies(request);
+	if(cookies['session.id']){
+		id=cookies['session.id'];
+	}
+
+	if(checkUserID(cookies['session.id'])){
+		state.authenticatedUsername = cookies['session.user'];
+		return true;
+	}
+
+	var headers = request.headers;
+
 	var header = request.headers.authorization || "";
 	if(!header && state.allowAnon) {
 		// If there's no header and anonymous access is allowed then we don't set authenticatedUsername
 		return true;
 	}
-	var token = header.split(/\s+/).pop() || "",
-		auth = $tw.utils.base64Decode(token),
-		parts = auth.split(/:/),
-		incomingUsername = parts[0],
-		incomingPassword = parts[1];
+
 	// Check that at least one of the credentials matches
 	var matchingCredentials = this.credentialsData.find(function(credential) {
-		return credential.username === incomingUsername && credential.password === incomingPassword;
+		return credential.username === headers.authuser && credential.password === headers.authpass;
 	});
+
+	//console.log("matching? "+matchingCredentials);
 	if(matchingCredentials) {
+		//console.log("matching! "+ headers.authuser);
+		//create new uuid to authenticate the client
+		id=uuid.v4();
+		authenticateUser(id,  timestamp, headers.authuser);
+		//send new id as content to the client
+		response.end(id);
 		// If so, add the authenticated username to the request state
-		state.authenticatedUsername = incomingUsername;
+		state.authenticatedUsername = headers.authuser;
 		return true;
 	} else {
-		// If not, return an authentication challenge
-		response.writeHead(401,"Authentication required",{
-			"WWW-Authenticate": 'Basic realm="Please provide your username and password to login to ' + state.server.servername + '"'
-		});
-		response.end();
+		if(headers.authuser || headers.authpass){
+			response.writeHead(406,"Wrong Password");
+			response.end();
+			return false;
+		}
+
+		id=uuid.v4();
+		timestamp = date.getTime();
+
+		var appDir = path.resolve('core/modules/server/authenticators/');
+		//var resp=fs.readFileSync(appDir+'/template.html', 'utf8');
```

Comment:
You can read the template directly from the tiddler using something like `$tw.wiki.getTiddlerText(title)` where title is the title of the tiddler containing the template.